src/py/flwr/common/telemetry.py

```diff
@@ -58,6 +59,42 @@ def log(msg: Union[str, Exception]) -> None:
     logging.getLogger(LOGGER_NAME).log(LOGGER_LEVEL, msg)
 
 
+def _get_home() -> Path:
+    return Path().home()
+
+
+def _get_source_id() -> str:
+    """Get existing or new source ID."""
+    source_id = "unavailable"
+    # Check if .flwr in home exists
+    try:
+        home = _get_home()
+    except RuntimeError:
+        # If the home directory can’t be resolved, RuntimeError is raised.
+        return source_id
+
+    flwr_dir = home.joinpath(".flwr")
+    # Create .flwr directory if it does not exist yet.
+    try:
+        flwr_dir.mkdir(parents=True, exist_ok=True)
+    except PermissionError:
+        return source_id
+
+    source_file = flwr_dir.joinpath("source")
+
+    # If no source_file exists create one and write it
+    if not source_file.exists():
+        try:
+            source_file.touch(exist_ok=True)
+            source_file.write_text(str(uuid.uuid4()), encoding="utf-8")
+        except PermissionError:
+            return source_id
+
+    source_id = source_file.read_text(encoding="utf-8").strip()
```

Comment:
Should we verify if the string we read is a valid UUID?

Reply:
I think it's better to send something like an invalid UUID rather than "unavailable". I would not validate it, therefore.